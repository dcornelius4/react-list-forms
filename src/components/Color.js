import React from 'react';
import PropTypes from 'prop-types';

const hexColor = color => {
  return `${color.toString(16).padStart(2, 0)}`.toUpperCase();
};

const hexRGB = (rgb) => {
  return `#${hexColor(rgb.red)}${hexColor(rgb.green)}${hexColor(rgb.blue)}`;
};

function Color({ name, rgb }) {
  const hex = hexRGB(rgb);

  const colorStyle = {
    display: 'flex',
    backgroundColor: hex,
    width: '1em',
    height: '1em'
  };

  return (
    <dl>
      <dt>Name</dt>
      <dd>{name || hex} <div style={colorStyle}></div></dd>

      <dt>RGB</dt>
      <dd>
        <p>Red - {rgb.red}</p>
        <p>Green - {rgb.green}</p>
        <p>Blue - {rgb.blue}</p>
      </dd>
    </dl>
  );
}

Color.propTypes = {
  name: PropTypes.string,
  rgb: PropTypes.shape({
    red: PropTypes.number.isRequired,
    green: PropTypes.number.isRequired,
    blue: PropTypes.number.isRequired
  })
};

export default Color;
