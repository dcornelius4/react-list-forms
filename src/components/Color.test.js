import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('Color Component', () => {
  it('renders a color', () => {
    const rgb = {
      red: 255,
      green: 0,
      blue: 0
    };

    const wrapper = shallow(<Color name="Red" rgb={rgb} />);
    expect(wrapper).toMatchSnapshot();
  });
});
