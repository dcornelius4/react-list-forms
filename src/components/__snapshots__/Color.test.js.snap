// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Component renders a color 1`] = `
<dl>
  <dt>
    Name
  </dt>
  <dd>
    Red
     
    <div
      style={
        Object {
          "backgroundColor": "#FF0000",
          "display": "flex",
          "height": "1em",
          "width": "1em",
        }
      }
    />
  </dd>
  <dt>
    RGB
  </dt>
  <dd>
    <p>
      Red - 
      255
    </p>
    <p>
      Green - 
      0
    </p>
    <p>
      Blue - 
      0
    </p>
  </dd>
</dl>
`;
